tests/governance_js.py

```diff
@@ -333,38 +308,105 @@ def test_operator_proposals_and_votes(network, args):
 def test_actions(network, args):
     node = network.find_random_node()
 
-    with node.client(None, "member0") as c:
-        valid_set_member_data = proposal(
-            action(
-                "set_member_data",
-                member_id=f"{network.consortium.get_member_by_local_id('member0').service_id}",
-                member_data={"is_admin": True},
-            )
-        )
+    # Rekey ledger
+    network.consortium.rekey_ledger(node)
 
-        r = c.post("/gov/proposals.js", valid_set_member_data)
-        assert r.status_code == 200, r.body.text()
+    # Add new user twice (with and without user data)
+    new_user_local_id = "js_user"
+    new_user = network.create_user(new_user_local_id, args.participants_curve)
+    LOG.info(f"Adding new user {new_user.service_id}")
 
-        valid_rekey_ledger = proposal(action("rekey_ledger"))
-        r = c.post("/gov/proposals.js", valid_rekey_ledger)
-        assert r.status_code == 200, r.body.text()
+    user_data = None
+    network.consortium.add_user(node, new_user.local_id, user_data)
 
-        valid_service_open = proposal(action("transition_service_to_open"))
-        r = c.post("/gov/proposals.js", valid_service_open)
-        assert r.status_code == 200, r.body.text()
+    user_data = {"foo": "bar"}
+    network.consortium.add_user(node, new_user.local_id, user_data)
 
-        new_user_local_id = "js_user"
-        new_user = network.create_user(new_user_local_id, args.participants_curve)
-        LOG.info(f"Adding new user {new_user.service_id}")
-        with open(
-            os.path.join(network.common_dir, f"{new_user_local_id}_cert.pem"), "r"
-        ) as cert:
-            valid_new_user = proposal(
-                action("set_user", cert=cert.read(), user_data={"is_admin": True})
-            )
-        r = c.post("/gov/proposals.js", valid_new_user)
+    with node.client(new_user.local_id) as c:
+        r = c.post("/app/log/private", {"id": 0, "msg": "JS"})
         assert r.status_code == 200, r.body.text()
-    return network
+
+    # Set user data
+    network.consortium.set_user_data(
+        node, new_user.service_id, user_data={"user": "data"}
+    )
+    network.consortium.set_user_data(node, new_user.service_id, user_data=None)
+
+    # TODO: Fail!
+    # Remove user
+    # network.consortium.remove_user(node, new_user.service_id)
+
+    # with node.client(new_user.local_id) as c:
+    #     r = c.get("/app/log/private")
+    #     assert r.status_code == 401, r.body.text()
+
+    # Set member data
+    network.consortium.set_member_data(
```

Comment:
Aren't there existing tests for these proposals?

Reply:
They are. But they are in other test files which also call other actions that would have failed before reaching the interesting test. I'll remove these once we have everything working!